docs/topics/development/search.rst

```diff
@@ -0,0 +1,92 @@
+.. _search:
+
+============================
+How does search on AMO work?
+============================
+
+.. note::
+
+  This is documenting our current state of how search is implemented in addons-server.
+  We will be using this to plan future improvements so please note that we are
+  aware that the process written below is not perfect and has bugs here and there.
+
+  Please see https://github.com/orgs/mozilla/projects/17#card-10287357 for more planning.
+
+
+General structure
+=================
+
+Our search contains these data types:
+
+ * addons (`Add-ons indexer / serializer <https://github.com/mozilla/addons-server/blob/master/src/olympia/addons/indexers.py#L22-L379>`_)
+
+  - versions (`Indexer / Serializer <https://github.com/mozilla/addons-server/blob/master/src/olympia/addons/indexers.py#L215-L237>`_)
+
+   + files
+   + compatibility information
+
+  - authors
+  - previews (image links)
+  - translations (`Translations mapping generation <https://github.com/mozilla/addons-server/blob/master/src/olympia/amo/indexers.py#L40-L136>`_)
+  - … various other add-on related properties
+
+ * add-on compatibility reports
+
+Our search can be reached either via the API through :ref:`/addons/search/ <addon-search>` or :ref:`/addons/autocomplete/ <addon-autocomplete>` which are used by our addons-frontend as well as via our legacy pages (used much less).
+
+Both use similar filtering and scoring code. For legacy reasons they're not identical though, we should try to focus on our API-based search though since the legacy search will be removed once support for Thunderbird and Seamonkey will be moved to a new platform.
+
+The legacy search uses ElasticSearch to query the data and then requests the actual model objects from the database. The newer API-based search only hits ElasticSearch and uses data directly stored from ES which is much more efficient.
+
+
+Flow of a search query through AMO
+==================================
+
+Let's assume we search on addons-frontend (not legacy) the search query hits the API and get's handled by `AddonSearchView` which directly queries ElasticSearch and doesn't involve the database at all.
+
+There are a few filters that are described in the :ref:`/addons/search/ docs <addon-search>` but most of them are not very relevant for raw search queries. Examplex are filters by guid, platform, category or add-on type.
+
+Much more relevant for raw add-on searches (And this is primarily used when you use the search on the frontend) is `SearchQueryFilter`.
+
+It composes various rules to define a more or less usable ranking:
+
+
+Primary rules
```

Comment:
This section probably deserves a sub header, and some more backticks for things like `param=value` things.